de.dlr.sc.virsat.model.extension.fdir.ui/src/de/dlr/sc/virsat/model/extension/fdir/ui/diagram/ft/features/faultTreeNodes/FaultTreeNodeLayoutFeature.java

```diff
@@ -109,7 +112,16 @@ public boolean layout(ILayoutContext context) {
 			}
 
 			anythingChanged |= layoutGa(nameText, 0, PADDING_Y, width, nameHeight);
-			anythingChanged |= layoutGa(containerGa, containerGa.getX(), containerGa.getY(), width, height);
+
+
+			int x = containerGa.getX();
+			int y = containerGa.getY();
+
+			// Set top left diagram border margins of fault tree nodes to prevent context menu clipping
+			x = x < TOP_LEFT_X_BORDER_MARGIN ? TOP_LEFT_X_BORDER_MARGIN : x;
+			y = y < TOP_LEFT_Y_BORDER_MARGIN ? TOP_LEFT_Y_BORDER_MARGIN : y;

```

Comment:
I am not sure this approach works, since it only modifies a single node.
What happens with the other auto layouted nodes? Is their position also then updated correctly to reflect that the parent node moved?

Reply:
It seems that subsequent nodes get aligned properly with the first node.

Reply:
Okay that surprises me but cool.